src/main/java/game/Actions.java

```diff
@@ -1,22 +1,28 @@
 package game;
 
+import java.util.ArrayList;
 import java.util.List;
 import java.util.Map;
 import java.util.Set;
 
 public class Actions implements Interactable {
 
     private int id;
-    private Map<String, Effect> effects;
+    private Map<String, Effects> commandEffectsMap;
+
+    public Actions(int id, Map<String, Effects> commandEffectsMap) {
+        this.id = id;
+        this.commandEffectsMap = commandEffectsMap;
+    }
 
     @Override
-    public boolean onCommand(Command command, Callback callback) {
+    public boolean handleCommand(Command command, Callback callback) {
         if (hasCommand(command.getAction())) {
-            Effect effect = getEffect(command.getAction());
-            if (effect != null) {
-                effect.apply(command.getGame());
+            Effects effects = getEffects(command.getAction());
+            if (effects != null) {
+                effects.apply(command.getGame(), callback);
```

Comment:
Good idea to include here the implementation of a pass through callback, extending the possibilities of your command handling.